scripts/vet.sh

```diff
@@ -163,7 +163,7 @@ XXXXX gRPC internal usage deprecation errors:
 BalancerAttributes is deprecated:
 CredsBundle is deprecated:
 internal.Logger is deprecated:
-Metadata is deprecated: use Attributes instead.
+Metadata is deprecated:
```

Comment:
Why was this needed? The deprecation message is still the same: https://github.com/grpc/grpc-go/blob/6f41085574082e83c928771cd494613568fd8a38/resolver/resolver.go#L119

Reply:
The new deprecation is coming from [envoy/config/rbac/v3/rbac.proto](
https://github.com/envoyproxy/envoy/blob/e294469598d0980a792221ce712eab14cc6511ab/api/envoy/config/rbac/v3/rbac.proto#L269-L272).

I think we should add a new entry instead of relaxing this pattern to avoid accidentally hiding other deprecation warnings in the future.
```
GetMetadata is deprecated: Marked as deprecated in envoy/config/rbac
```

Reply:
> By the way, would you be open to use golangci-lint to simplify linters maintenance ?

I think we've been able to achieve what we need using vet and staticcheck directly combined with regexes, so there wasn't a need to add a new dependency. If you think think using `golangci-lint` has significant benefits, please do file an issue and we can discuss this.